api/kyverno/v1/rule_types.go

```diff
@@ -266,7 +269,7 @@ func (r *Rule) ValidateMatchExcludeConflict(path *field.Path) (errs field.ErrorL
 		}
 		return errs
 	}
-	if datautils.DeepEqual(r.ExcludeResources, MatchResources{}) {
+	if r.ExcludeResources == nil || datautils.DeepEqual(*r.ExcludeResources, MatchResources{}) {
```

Comment:
No need to check `r.ExcludeResources == nil` as it returns directly.

Reply:
Fixed